default.nix

```diff
@@ -2,6 +2,17 @@
 
 with import <nixpkgs> {};
 
+let why3_local =
+  why3.overrideAttrs (o : rec {
+    version = "1.5.1";
+    src = fetchurl {
+      url = "https://why3.gitlabpages.inria.fr/releases/${o.pname}-${version}.tar.gz";
+      sha256 = "sha256-vNR7WeiSvg+763GcovoZBFDfncekJMeqNegP4fVw06I=";
+    };
+  });
+in
+let why3 = why3_local; in
+
```

Comment:
Why using an intermediate `why3_local` variable?

Reply:
The `let` construction is *always* recursive in the nix language.